opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -159,7 +159,7 @@ namespace Opm
                            const WellState& well_state,
                            DeferredLogger& /* deferred_logger */)
     {
-        const bool stop_or_zero_rate_target = this->stopppedOrZeroRateTarget(summary_state, well_state);
+        const bool stop_or_zero_rate_target = this->stoppedOrZeroRateTargetIndividual(summary_state, well_state);
```

Comment:
I don't think setting the `WQTotal` of first segment explicitly to zero for zero-target wells is strictly necessary below, so have left out the checking for zero targets from groups here.

Reply:
It can be debatable. If we want to be a little more systematical, we probably want to remove this `stop_or_zero_rate_target` by all if it is not necessary. 

I think it is not a problem when it is with the correct sign, and it might be a problem when the sign of the `WQTotal` switches, which is easy for the wells without a zero rate target. 

So far we do not know how to handle wells with the opposite production/injection direction.

Reply:
> I think it is not a problem when it is with the correct sign, and it might be a problem when the sign of the WQTotal switches, which is easy for the wells without a zero rate target.

I beleive `WQTotal` is not allowed to switch sign in the Newton update, so it either has the right sign or is zero.